aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelTree.java

```diff
@@ -160,14 +146,34 @@ private void recurseTree(
         }
     }
 
-    private Cursor queryAll(
-            @NonNull String tableName,
-            @NonNull QueryOptions options
-    ) throws DataStoreException {
-        final ModelSchema schema = registry.getModelSchemaForModelClass(tableName);
-        final SqlCommand sqlCommand = commandFactory.queryFor(schema, options);
-        final String rawQuery = sqlCommand.sqlStatement();
-        final String[] bindings = sqlCommand.getBindingsAsArray();
-        return database.rawQuery(rawQuery, bindings);
+    private Cursor queryChildren(
+            @NonNull String childTable,
+            @NonNull String childIdField,
+            @NonNull String parentIdField,
+            @NonNull Collection<String> parentIds
+    ) {
+        // SELECT <child_id>, <parent_id> FROM <child_table> WHERE <parent_id> IN (<id_1>, <id_2>, ...)
+        String queryString = String.valueOf(SqlKeyword.SELECT) +
+                SqlKeyword.DELIMITER +
+                Wrap.inBackticks(childIdField) +
+                SqlKeyword.SEPARATOR +
+                Wrap.inBackticks(parentIdField) +
+                SqlKeyword.DELIMITER +
+                SqlKeyword.FROM +
+                SqlKeyword.DELIMITER +
+                Wrap.inBackticks(childTable) +
```

Comment:
Maybe we can create a little help method for these repeated groupings of tokens, like
```java
from(childTable)
```

Sort of like a domain specific language for SQL, but using static factory methods?

Overall I feel like our use of these consts is making it harder to see what the actual SQL commands will look like, when done.

We can make more, like:

```java
where(parentIdField).in(parentIds)
```

Reply:
this is a fantastic idea! I think it deserves its own PR because there is a bunch of similar code in `SQLiteCommandFactory` that will be made much cleaner with the addition of this utility.